runtime/verutil/cfrerr.c

```diff
@@ -57,6 +57,32 @@ getJ9CfrErrorNormalMessage(J9PortLibrary* portLib, J9CfrError* error, const U_8*
 	return errorString;
 }
 
+const char*
+getJ9CfrErrorClassNameMessage(J9PortLibrary* portLib, J9CfrError* error, const U_8* className, UDATA classNameLength)
+{
+	PORT_ACCESS_FROM_PORT(portLib);
+	UDATA allocSize = 0;
+	char *errorString = NULL;
+	J9CfrConstantPoolInfo *info = &error->constantPool[error->errorCPIndex];
+	UDATA utf8NameLength = error->constantPool[info->slot1].slot1;
+	U_8* utf8NameBytes = error->constantPool[info->slot1].bytes;
+	UDATA thisClassNameLength = (NULL != className) ? classNameLength : error->constantPool[error->thisClassIndex].slot1;
+	const U_8* thisClassNameBytes = (NULL != className) ? className : error->constantPool[error->thisClassIndex].bytes;
+
+	/* J9NLS_CFR_ERR_CP_BAD_CLASS_NAME=The class name (errorNo=#%1$u) in the constant pool entry at index (#%2$u) is invalid; utf8name=%4$.*3$s, class=%6$.*5$s, offset=%7$u */
+	const char *template = j9nls_lookup_message(J9NLS_ERROR | J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_CFR_ERR_CP_BAD_CLASS_NAME,
+			"The class name (errorNo=#%u) in the constant pool entry at index (#%u) is invalid; utf8name=%.*s, class=%.*s, offset=%u");
+
+	allocSize = strlen(template) + utf8NameLength + thisClassNameLength + (MAX_INT_SIZE * 3);
+	errorString = j9mem_allocate_memory(allocSize, OMRMEM_CATEGORY_VM);
```

Comment:
Where is this freed?

Reply:
In our case, the calling path is as follows:
```
createROMClassFromClassFile (/runtime/bcutil/defineclass.c /calling j9mem_free_memory)
|--> buildVerifyErrorString
         |---> getJ9CfrErrorDetailMessageNoMethod
                       |---> getJ9CfrErrorClassNameMessage (calling j9mem_allocate_memory)
```

Following the existing implementation in allocating/releasing memory, the memory allocated in getJ9CfrErrorClassNameMessage will eventually be released in createROMClassFromClassFile  after use as follows:
```
createROMClassFromClassFile(...)
{
...
		case BCT_ERR_GENERIC_ERROR_CUSTOM_MSG: {
			/* default value for exceptionNumber (J9VMCONSTANTPOOL_JAVALANGCLASSFORMATERROR) assigned before switch */		
----> errorUTF = (U_8 *)buildVerifyErrorString(vm, (J9CfrError *)vm->dynamicLoadBuffers->classFileError, className, classNameLength);
			break;
		}
...
	} else {
		if (errorUTF == NULL) {
			J9_VM_FUNCTION(currentThread, setCurrentException)(currentThread, exceptionNumber, NULL);
		} else {
			PORT_ACCESS_FROM_JAVAVM(vm);
			J9_VM_FUNCTION(currentThread, setCurrentExceptionUTF)(currentThread, exceptionNumber, (const char*)errorUTF);
---> j9mem_free_memory(errorUTF);
		}
	}
```